src/write/column_chunk.rs

```diff
@@ -49,16 +50,37 @@ where
     }
     let mut bytes_written = offset - initial;
 
-    let column_chunk = build_column_chunk(&specs, descriptor)?;
+    let mut column_chunk = build_column_chunk(&specs, descriptor)?;
 
     // write metadata
-    let mut protocol = TCompactOutputProtocol::new(writer);
+    let mut protocol = TCompactOutputProtocol::new(&mut *writer);
+
+    #[cfg(feature = "bloom_filter")]
+    if let Some(bloom_filter_bitset) = bloom_filter_bitset {
+        column_chunk.meta_data.as_mut().unwrap().bloom_filter_offset =
+            Some(offset.try_into().unwrap());
+        let written = crate::bloom_filter::write_to_protocol(
+            &mut protocol,
+            bloom_filter_bitset.len().try_into().unwrap(),
+        )?;
+
+        bytes_written += u64::try_from(written).unwrap();
+    }
+
     bytes_written += column_chunk
         .meta_data
         .as_ref()
         .unwrap()
         .write_to_out_protocol(&mut protocol)? as u64;
 
+    std::mem::drop(protocol);
+
+    #[cfg(feature = "bloom_filter")]
```

Comment:
Would it make sense to have this be a run-time error, like we have for compression where we raise it run-time?

Reply:
done